crates/example-types/src/storage_types.rs

```diff
@@ -0,0 +1,67 @@
+use anyhow::{bail, Result};
+use async_lock::RwLock;
+use async_trait::async_trait;
+use hotshot_types::{
+    data::{DAProposal, VidDisperse},
+    message::Proposal,
+    traits::{node_implementation::NodeType, storage::Storage},
+};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Clone, Debug)]
+pub struct TestStorageState<TYPES: NodeType> {
+    vids: HashMap<TYPES::Time, Proposal<TYPES, VidDisperse<TYPES>>>,
+    das: HashMap<TYPES::Time, Proposal<TYPES, DAProposal<TYPES>>>,
+}
+
+impl<TYPES: NodeType> Default for TestStorageState<TYPES> {
+    fn default() -> Self {
+        Self {
+            vids: HashMap::new(),
+            das: HashMap::new(),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct TestStorage<TYPES: NodeType> {
+    inner: Arc<RwLock<TestStorageState<TYPES>>>,
+
+    /// `should_return_err` is a testing utility to validate negative cases.
+    pub should_return_err: bool,
```

Comment:
Open to thoughts here. See the tests below for examples. I just need a cheesy "fail immediately" system that does not change the interface. The idea here is that I just want to make sure that any logical changes can catch a regression in the failure case for an `append`.

Reply:
Yeah this is fine, it's just a dummy type for our tests anyway